public/app/core/components/AppChrome/AppChromeService.tsx

```diff
@@ -94,6 +94,9 @@ export class AppChromeService {
   }
 
   public setReturnToPrevious = (returnToPrevious: ReturnToPreviousProps) => {
+    if (!returnToPrevious.href) {
+      returnToPrevious.href = locationService.getLocation().pathname;
+    }
```

Comment:
It's poor practice to mutate incoming objects like this because it can lead to subtle bugs that are hard to figure out.

But, I don't think the href on this method should be optional. I think inside the hook should be where it defaults to the current location.

Reply:
Could change `useReturnToPreviousInternal` to something like

```ts
export function useReturnToPreviousInternal() {
  const { chrome } = useGrafana();

  return useCallback(
    (title: string, href?: string) => {
      chrome.setReturnToPrevious({
        title: title,
        href: href || locationService.getLocation().pathname,
      });
    },
    [chrome]
  );
}
```

Reply:
Ok, I was doubting whether to do it in `AppChromeService` or the hook.